design/witx-type-representation.md

```diff
@@ -0,0 +1,86 @@
+# Witx Type Representation
+In Witx, each type corresponds to a lower-level type. For example, the `size` type resolves to a `u32`, the `errno` type resolves to an `Enum(u16)`, and more. Detailed below is each type, and what they resolve to in memory.
+
+# u8/u16/u32/u64/s64
+These are C-style `uint8_t`s, `uint16_t`s, `uint32_t`s, `uint64_t`s, and `int64_t`s. In the .NET world, they may be more recognizable as an `byte` (in C# this is an unsigned `byte` if you're coming from Java), `ushort`, `uint`, `ulong`, and `long`. These are typically little-endian, although it may not be necessary to know that.
```

Comment:
It may be worth pointing to https://github.com/WebAssembly/interface-types/blob/master/proposals/interface-types/Explainer.md#integers as the most authoritative reference for the signedness-aware integer types we have right now.

> in C# this is an unsigned `byte` if you're coming from Java

Is this parenthetical about C# or Java?

Reply:
done & removed that whole parenthetical part, using just the C types is good enough